src/test/java/com/agilethought/internship/sso/validator/user/UpdateUserValidatorTest.java

```diff
@@ -21,129 +21,99 @@
-                () -> {
-                    updateUserValidator.validate(mockUpdateRequest);
-                }
-        );
-        assertEquals(
-                "Email my_email@example.com is already in use, try with another one or login.",
-                thrownException.getMessage()
-        );
-
-    }
-
-    @Test
-    public void testUpdateUserByIdSuccessfullyWithNewEmail() {
-
-        User mockUpdateUser = new User();
-        mockUpdateUser.setId("1234");
-        mockUpdateUser.setEmail("new_unique_email@example.com");
-
-        when(repositoryApplication.findById(anyString()))
-                .thenReturn(Optional.of(mockUpdateUser));
-        doNothing().when(userDataValidator).validate(any());
-        when(repositoryApplication.existsByEmail(anyString())).thenReturn(false);
-        assertDoesNotThrow(
-                () -> {
-                    updateUserValidator.validate(mockUpdateUser);
-                }
-        );
-
-    }
-
-    @Test
-    public void testUpdateUserByIdWithSameEmail() {
-
-        User mockUpdateUser = new User();
-        mockUpdateUser.setId("1234");
-        mockUpdateUser.setEmail("my_same_email@example.com");
-
-        when(repositoryApplication.findById(anyString()))
-                .thenReturn(Optional.of(mockUpdateUser));
-        doNothing().when(userDataValidator).validate(any());
-        when(repositoryApplication.existsByEmail(anyString())).thenReturn(true);
-        assertDoesNotThrow(
-                () -> {
-                    updateUserValidator.validate(mockUpdateUser);
-                }
-        );
-    }
-
-}
+	@Mock
+	private RepositoryApplication repositoryApplication;
+
+	@Mock
+	private UserDataValidator userDataValidator;
+
+	@InjectMocks
+	private UpdateUserValidator updateUserValidator;
+
+	@Before
+	public void setup() {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void testUpdateUserByIdWithMissingId() {
+
+		User mockUpdateUser = new User();
+
+		Exception thrownException = assertThrows(BadRequestException.class, () -> {
+			updateUserValidator.validate(mockUpdateUser);
+		});
+		assertEquals("Required field id is missing.", thrownException.getMessage());
+	}
+
+	@Test
+	public void testUpdateUserByIdWithIdNotFound() {
+
+		User mockUpdateUser = new User();
+		mockUpdateUser.setId("mockedId");
+
+		when(repositoryApplication.findById(anyString())).thenReturn(Optional.empty());
+		Exception thrownException = assertThrows(NotFoundException.class, () -> {
+			updateUserValidator.validate(mockUpdateUser);
+		});
+		assertEquals("User was not found with the given id: mockedId", thrownException.getMessage());
+	}
+
+	@Test
+	public void testUpdateUserByIdWithEmailAlreadyTaken() {
+
+		// A user with email "current_email@example.com" tries
+		// to update its email to "my_email@example.com", but
+		// it's already taken by another user.
+		User mockUpdateRequest = new User();
+		mockUpdateRequest.setId("mockedUserToUpdateId");
```

Comment:
Do you have any constants class for the tests ?
Could be better if these values are in a constant class.

Reply:
The unit test classes where included as a migration from the login functionality that used to be in the SCE repository, and are not part of this specific current story, so I would suggest to establish a refactor later on to improve best practices implementation.